deploy/scripts/shared.sh

```
@@ -109,15 +96,18 @@ render_service() {
-            local dest_config="${CONFIG_READONLY_DIR}/flightctl-${service_name}/$(basename "$config_file")"
-            echo "copy config: ${config_file} -> ${dest_config}"
-            cp "$config_file" "${dest_config}"
-        fi
-    done
+    local config_dir="${source_dir}/flightctl-${service_name}/flightctl-${service_name}-config"
+    if [[ -d "$config_dir" ]]; then
+        for config_file in "$config_dir"/*; do
+            if [[ -f "$config_file" ]]; then
+                # Ensure config output directory exists
+                mkdir -p "${CONFIG_READONLY_DIR}/flightctl-${service_name}"
+                local dest_config="${CONFIG_READONLY_DIR}/flightctl-${service_name}/$(basename "$config_file")"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)